src/compressed_tensors/compressors/base.py

```diff
@@ -12,14 +12,22 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-from typing import Dict, Generator, Tuple, Union
```

Comment:
great refactor - I'd update the class docstring to give a little more information on the lifecycle and expected user interaction with the methods